modules/programs/starship.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.starship;
+
+  configFile = config:
+    pkgs.runCommand "config.toml"
+      {
+        buildInputs = [ pkgs.remarshal ];
+        preferLocalBuild = true;
+        allowSubstitutes = false;
+      }
+      ''
+        remarshal -if json -of toml \
+          < ${pkgs.writeText "config.json" (builtins.toJSON config)} \
+          > $out
+      '';
+in
+
+{
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.starship = {
+    enable = mkEnableOption "starship";
+
+    settings = mkOption {
+      type = types.attrs;
+      default = {};
+      description = ''
+        Configuration written to
+        <filename>~/.config/starship.toml</filename>.
+        </para><para>
+        See <link xlink:href="https://starship.rs/config/" /> for the full list
+        of options.
+      '';
+    };
+
+    enableBashIntegration = mkOption {
+      default = true;
```

Comment:
Please don't enable these by default. Let the user enable these option explicitly.

You can use `mkEnableOption` here as well

Reply:
@Gerschtli Having these enabled mirrors other similar modules, e.g., fzf, direnv, … The idea is that if you enable the starship module then you typically would like to have starship enabled in the shells you use. This option is more when you specifically want to disable starship in zsh and use it in bash, for example.

Reply:
Ah okay, seems reasonable. I thought these options were disabled by default in similar modules like fzf.